otx/algorithms/detection/adapters/mmdet/configurer.py

```diff
@@ -0,0 +1,757 @@
+"""Base configurer for mmdet config."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import importlib
+import json
+import os
+from typing import Any, Dict
+
+import numpy as np
+import torch
+from mmcv.runner import CheckpointLoader
+from mmcv.utils import Config, ConfigDict
+from torch import distributed as dist
+
+from otx.algorithms.common.adapters.mmcv.utils import (
+    align_data_config_with_recipe,
+    build_dataloader,
+    build_dataset,
+    patch_default_config,
+    patch_fp16,
+    patch_persistent_workers,
+    patch_runner,
+    update_basic_hooks,
+)
+from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
+    recursively_update_cfg,
+    update_or_add_custom_hook,
+)
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.detection.adapters.mmdet.utils import (
+    patch_datasets,
+    patch_evaluation,
+)
+
+logger = get_logger()
+
+
+# pylint: disable=too-many-public-methods
+class DetectionConfigManager:
+    """Patch config to support otx train."""
+
+    def __init__(self):
+        self.task_adapt_type = None
+        self.task_adapt_op = "REPLACE"
+        self.org_model_classes = []
+        self.model_classes = []
+        self.data_classes = []
+
+    # pylint: disable=too-many-arguments
+    def configure(
+        self,
+        cfg,
+        model_ckpt,
+        data_cfg,
+        training=True,
+        subset="train",
+        ir_options=None,
+        data_classes=None,
+        model_classes=None,
+    ):
+        """Create MMCV-consumable config from given inputs."""
+        logger.info(f"configure!: training={training}")
+
+        self.base_configure(cfg, data_cfg, data_classes, model_classes)
+        self.configure_device(cfg)
+        self.configure_model(cfg, ir_options)
+        self.configure_ckpt(cfg, model_ckpt)
+        self.configure_data(cfg, training, data_cfg)
+        self.configure_regularization(cfg, training)
+        self.configure_task(cfg, training)
+        self.configure_hook(cfg)
+        self.configure_samples_per_gpu(cfg, subset)
+        self.configure_fp16_optimizer(cfg)
+        self.configure_compat_cfg(cfg)
+        return cfg
+
+    def base_configure(self, cfg, data_cfg, data_classes, model_classes):
```

Comment:
```suggestion
    def configure_base(self, cfg, data_cfg, data_classes, model_classes):
```

Reply:
It is base configurations step, not configure base.

Reply:
what is the difference between the two? I'm just wondering why this function use `base_configure` although all other configure functions used `configure_****` name.

```python 3
self.base_configure(cfg, data_cfg, data_classes, model_classes)
self.configure_device(cfg)
self.configure_model(cfg, ir_options)
self.configure_ckpt(cfg, model_ckpt)
self.configure_data(cfg, training, data_cfg)
self.configure_regularization(cfg, training)
self.configure_task(cfg, training)
self.configure_hook(cfg)
self.configure_samples_per_gpu(cfg, subset)
self.configure_fp16_optimizer(cfg)
self.configure_compat_cfg(cfg)
```

Reply:
For example configure_device is configuring device setting, but base_configure do not contain exact object for configuration. Therefore I make this function as a base configuration function before configuration. However if you think this is not make sense, then I'll change of course. I want to explain my point

Reply:
How about `configure_base`? or `configure_init`?